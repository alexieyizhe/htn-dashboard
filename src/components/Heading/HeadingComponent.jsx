import styled from "styled-components";

const Heading = styled.div`
  font-size: 2em;
  font-weight: 500;
  color: ${props => props.main ? props.theme.colors.black : props.theme.colors.grey};
  margin-bottom: ${props => props.noMargin ? 0 : '0.5em'};
`;

export default Heading;
