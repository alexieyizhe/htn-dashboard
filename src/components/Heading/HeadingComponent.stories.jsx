import React from "react";
import { storiesOf } from "@storybook/react";
import Heading from "./HeadingComponent";


storiesOf("HeaderButtons", module)
  .add("Default", () => <Heading>Default</Heading>)
  .add("Main", () => <Heading main>Main Heading</Heading>)
  .add("No margin", () => <Heading noMargin>No Margin</Heading>)
  .add("Main & No Margin", () => <Heading noMargin main>Both!</Heading>);
