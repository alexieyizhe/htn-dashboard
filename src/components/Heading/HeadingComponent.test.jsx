/* eslint-disable no-undef */
import React from "react";

import Heading from "./HeadingComponent";

describe("Heading", () => {
  it("renders correctly", () => {
    const component = mountWithTheme(<Heading>Test Heading</Heading>);
    expect(component).toMatchSnapshot();
  })
})
