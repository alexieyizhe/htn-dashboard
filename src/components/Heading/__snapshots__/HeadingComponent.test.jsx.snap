// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders correctly 1`] = `
.c0 {
  font-size: 2em;
  font-weight: 500;
  color: #AAAAAA;
  margin-bottom: 0.5em;
}

<styled.div>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  font-size: 2em;
  font-weight: 500;
  color: ",
            [Function],
            ";
  margin-bottom: ",
            [Function],
            ";
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    >
      Test Heading
    </div>
  </StyledComponent>
</styled.div>
`;
